gradio/brotli_middleware.py

```diff
@@ -91,6 +95,18 @@ def _is_handler_excluded(self, scope: Scope) -> bool:
 
         return any(pattern.search(handler) for pattern in self.excluded_handlers)
 
+    # explicitly handle html, js, css, json via a whitelist. woff2 files are already compressed.
+    # we don't want to compress binary files as they do not benefit much and it can cause bugs
+    def _is_compressible_file_type(self, scope: Scope) -> bool:
+        """Check if the requested file has a compressible file extension."""
+        path = scope.get("path", "")
+        compressible_extensions = {".html", ".htm", ".js", ".css", ".json"}
```

Comment:
Should we make this list exhaustive for all text file types? ex: add `.txt`, `.md`, `.xhtml` and probably more.

Reply:
yeah we should probably add those, although I'm wary of adding too many that we haven't tested in case people load them in weird ways.